libfoundation/src/foundation-java.cpp

```diff
@@ -0,0 +1,234 @@
+/* Copyright (C) 2003-2016 LiveCode Ltd.
+
+This file is part of LiveCode.
+
+LiveCode is free software; you can redistribute it and/or modify it under
+the terms of the GNU General Public License v3 as published by the Free
+Software Foundation.
+
+LiveCode is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received a copy of the GNU General Public License
+along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
+
+#include <foundation.h>
+#include <foundation-auto.h>
+
+#include "foundation-private.h"
+
+#ifdef TARGET_PLATFORM_MACOS_X
+#include <JavaVM/jni.h>
+#elif TARGET_SUBPLATFORM_ANDROID
+#include <jni.h>
+extern JNIEnv *MCJavaGetThreadEnv();
+extern JNIEnv *MCJavaAttachCurrentThread();
+extern void MCJavaDetachCurrentThread();
+#endif
+
+#if defined(TARGET_PLATFORM_MACOS_X) || defined(TARGET_SUBPLATFORM_ANDROID)
+#define TARGET_SUPPORTS_JAVA
+#endif
+
+#ifdef TARGET_SUPPORTS_JAVA
+static JNIEnv *s_env;
+static JavaVM *s_jvm;
+#endif
+
+bool __MCJavaInitialize()
+{
+#ifdef TARGET_PLATFORM_MACOS_X
+    JavaVMInitArgs vm_args;
+    vm_args.version = JNI_VERSION_1_6;
+    JNI_GetDefaultJavaVMInitArgs(&vm_args);
+
+    JavaVMOption* options = new JavaVMOption[1];
+    options[0].optionString = "-Djava.class.path=/usr/lib/java";
+    
+    vm_args.nOptions = 1;
+    vm_args.options = options;
+    vm_args.ignoreUnrecognized = false;
+    
+    JNI_CreateJavaVM(&s_jvm, (void **)&s_env, &vm_args);
```

Comment:
I wonder if the best thing to do would be to allow it to fail, and just keep a variable around like `s_java_initialised` so that everything can just return false if it failed